src/inspector_js_api.cc

```diff
@@ -125,18 +101,22 @@ class JSBindingsConnection : public AsyncWrap {
     ASSIGN_OR_RETURN_UNWRAP(&session, info.Holder());
     CHECK(info[0]->IsString());
 
-    session->CheckIsCurrent();
-    Agent* inspector = env->inspector_agent();
-    inspector->Dispatch(ToProtocolString(env->isolate(), info[0])->string());
+    if (session->session_)
+      session->session_->Dispatch(
+          ToProtocolString(env->isolate(), info[0])->string());
   }
 
   size_t self_size() const override { return sizeof(*this); }
 
  private:
-  JSBindingsSessionDelegate delegate_;
+  std::unique_ptr<InspectorSession> session_;
   Persistent<Function> callback_;
 };
 
+static bool InspectorEnabled(Environment* env) {
+  Agent* agent = env->inspector_agent();
+  return !!agent->io() || agent->HasConnectedSessions();
```

Comment:
Can you do an explicit nullptr check instead of coercing to bool?

Reply:
Done